src/state/slices/foxEthSlice/selectors.ts

```diff
@@ -0,0 +1,320 @@
+import { createSelector } from '@reduxjs/toolkit'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { CHAIN_NAMESPACE, ethAssetId, fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import keys from 'lodash/keys'
+import { createCachedSelector } from 're-reselect'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { toBaseUnit } from 'lib/math'
+import type { ReduxState } from 'state/reducer'
+import { createDeepEqualOutputSelector } from 'state/selector-utils'
+import { selectAssets } from 'state/slices/assetsSlice/selectors'
+import { selectMarketData } from 'state/slices/marketDataSlice/selectors'
+
+import type { AccountSpecifier } from '../accountSpecifiersSlice/accountSpecifiersSlice'
+import { foxEthLpAssetId } from './constants'
+import type { FoxEthLpEarnOpportunityType, FoxFarmingEarnOpportunityType } from './foxEthCommon'
+
+type ParamFilter = {
+  accountAddress?: string
+  contractAddress?: string
+}
+type OptionalParamFilter = {
+  accountAddress?: string
+  contractAddress?: string
+}
+
+type ParamFilterKey = keyof ParamFilter
+type OptionalParamFilterKey = keyof OptionalParamFilter
+
+const selectParamFromFilter =
```

Comment:
We'll want to abstract this pattern to its own helper file, but since there might be requested changes in this PR, didn't tackle it as a stack yet